modules/apps/content-dashboard/content-dashboard-web/src/test/java/com/liferay/content/dashboard/web/internal/item/JournalArticleContentDashboardItemTest.java

```diff
@@ -160,6 +176,181 @@ public void testGetAssetTags() {
+	}
+
+	@Test
+	public void testGetDefaultContentDashboardItemActionWithApprovedAndDraftStatusAndNotOwnerUser()
+		throws Exception {
+
+		JournalArticle journalArticle1 = _getJournalArticle();
+
+		Mockito.when(
+			journalArticle1.getStatus()
+		).thenReturn(
+			WorkflowConstants.STATUS_APPROVED
+		);
+
+		JournalArticle journalArticle2 = _getJournalArticle();
+
+		Mockito.when(
+			journalArticle2.getStatus()
+		).thenReturn(
+			WorkflowConstants.STATUS_DRAFT
+		);
+
+		Mockito.when(
+			journalArticle2.getVersion()
+		).thenReturn(
+			1.1
+		);
+
+		JournalArticleContentDashboardItem journalArticleContentDashboardItem =
+			new JournalArticleContentDashboardItem(
+				null, null,
+				_getContentDashboardItemActionProviderTracker(
+					_getContentDashboardItemActionProvider(
+						ContentDashboardItemAction.Type.VIEW,
+						"http://localhost:8080/view"),
+					_getContentDashboardItemActionProvider(
+						ContentDashboardItemAction.Type.EDIT,
+						"http://localhost:8080/edit")),
+				null, null, null, journalArticle1, new LanguageImpl(),
+				journalArticle2, new PortalImpl());
+
+		ContentDashboardItemAction contentDashboardItemAction =
+			journalArticleContentDashboardItem.
+				getDefaultContentDashboardItemAction(
+					_getHttpServletRequest(RandomTestUtil.randomLong()));
+
+		Assert.assertEquals("viewURL", contentDashboardItemAction.getURL());
+	}
+
+	@Test
+	public void testGetDefaultContentDashboardItemActionWithApprovedAndDraftStatusAndOwnerUser()
+		throws Exception {
+
+		JournalArticle journalArticle1 = _getJournalArticle();
+
+		Mockito.when(
+			journalArticle1.getStatus()
+		).thenReturn(
+			WorkflowConstants.STATUS_APPROVED
+		);
+
+		Mockito.when(
+			journalArticle1.getUserId()
+		).thenReturn(
+			RandomTestUtil.randomLong()
+		);
+
+		JournalArticle journalArticle2 = _getJournalArticle();
+
+		Mockito.when(
+			journalArticle2.getStatus()
+		).thenReturn(
+			WorkflowConstants.STATUS_DRAFT
+		);
+
+		Mockito.when(
+			journalArticle2.getUserId()
+		).thenReturn(
+			RandomTestUtil.randomLong()
+		);
+
+		Mockito.when(
+			journalArticle2.getVersion()
+		).thenReturn(
+			1.1
+		);
+
+		JournalArticleContentDashboardItem journalArticleContentDashboardItem =
+			new JournalArticleContentDashboardItem(
+				null, null,
+				_getContentDashboardItemActionProviderTracker(
+					_getContentDashboardItemActionProvider(
+						ContentDashboardItemAction.Type.VIEW,
+						"http://localhost:8080/view"),
+					_getContentDashboardItemActionProvider(
+						ContentDashboardItemAction.Type.EDIT,
+						"http://localhost:8080/edit")),
+				null, null, null, journalArticle1, new LanguageImpl(),
+				journalArticle2, new PortalImpl());
+
+		ContentDashboardItemAction contentDashboardItemAction =
+			journalArticleContentDashboardItem.
+				getDefaultContentDashboardItemAction(
+					_getHttpServletRequest(journalArticle2.getUserId()));
+
+		Assert.assertEquals("editURL", contentDashboardItemAction.getURL());
```

Comment:
Should it be "http://localhost:8080/edit"?

Reply:
Bad last minuute replacement 🤦